MicroPython/src/common/utils.py

```diff
@@ -171,26 +171,28 @@ def get_wifi_and_cloud_handlers(sync_time: bool = False) -> (WirelessConnectionC
         except Exception:
             logging.error("Error in disconnecting WiFi controller")
 
-        logging.debug("RESETTING BOARD")
-        machine.reset()
+        logging.debug("sleep({})".format(config.cfg.data_publishing_period_in_ms))
+        machine.deepsleep(config.cfg.data_publishing_period_in_ms)
 
     return wireless_controller, mqtt_communicator
 
 
-def connect_to_wifi(wireless_controller: WirelessConnectionController, sync_time: bool = False) -> None:
+def connect_to_wifi(wireless_controller: WirelessConnectionController, wifi_credentials: list[dict],
+                    sync_time: bool = False) -> None:
     """
     Connects ESP to wifi.
     :param wireless_controller: Wifi handler
+    :param wifi_credentials: list of ssid & password pairs
     :param sync_time: flag if time is synced.
     :return: None
     """
     logging.debug("utils.py/connect_to_wifi({})".format(sync_time))
-    wireless_controller.setup_station(access_points=config.cfg.access_points)
+    wireless_controller.setup_station(access_points=wifi_credentials)
 
     try:
         wireless_controller.configure_station()
     except Exception as e:
-        logging.info("Failed to connect to wifi {}".format(e))
+        logging.info(e)
```

Comment:
Why did You remove it? Please restore it as it gives more information than just printing an exception.